lib/potluck.rb

```diff
@@ -0,0 +1,37 @@
+require './lib/dish.rb'
+require 'pry'
+
+class Potluck
+
+  attr_accessor :date,
+                :dishes,
+                :name,
+                :category,
+                :dishes_by_category
+
+  def initialize(date)
+    @date = date
+    @dish = Dish.new(name, category)
+    @name = name
+    @category = category
+    @dishes = []
+    @dishes_by_category = Hash.new([])
+  end
+
+  def add_dish(dish)
+    @dishes << dish
+    add_to_hash(dish)
+  end
+  def add_to_hash(dish)
+    if @dishes_by_category.has_key?[dish.category]
+      @dishes_by_category[dish.category] << dish
+    else
+      @dishes_by_category[dish.category] = dish
+    end
```

Comment:
This totally does the trick! However, having a hash for dishes by category means that you then have to maintain it going forward. What happens if someone deletes their dish from the potluck (decides not to come)? Now you have to create a method to decrement the keys in the hash and also to remove a key if the value goes to 0.

Instead, consider creating a method `dishes_by_category` that will iterate over your `dishes` array to create this hash on the fly.